hack/verify-contextual-logging.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o nounset
+set -o pipefail
+# Don't set errexit because logcheck's messages are in stderr
+
+SHOW=${1:-}
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+# All subdirectories under ./pkg containing *.go files, _except_ in pkg/apis.
+# logcheck doesn't like that our main module doesn't reference pkg/apis.
+PKGS=$(cd "${SCRIPT_ROOT}"; find ./pkg -type f -name "*.go" -not -path "./pkg/apis/*" | sed -r 's|/[^/]+$||' | sort | uniq)
+
+
+errs=$(logcheck -check-contextual ${PKGS} 2>&1)
+
+if [[ ${SHOW} == "--show" ]]; then
+    echo "${errs}"
+    exit
+fi 
+
+err_cnt=$(echo "${errs}" | wc -l | xargs)
+
+if [[ ${err_cnt} -gt 386 ]]; then
```

Comment:
Instead of counting, since we have the actual error output, let's hold it in a file and diff it, so you can be more selective on what we are saying here. Also, would be cool to have a make target or script to update the file (when the count goes down)

Reply:
Yeah, that would be helpful.

Do we have a way to automatically check that file into git within our CI? Or do we just want to ask folks to run the script and include it when they do log changes?

Reply:
Just plop it into `hack/logcheck.out` and it should be checked in :)